bin/samlocal

```diff
@@ -78,7 +81,25 @@ def is_ecr_url(url: str) -> bool:
 is_ecr_url_orig = ecr_utils.is_ecr_url
 ecr_utils.is_ecr_url = is_ecr_url
 types.is_ecr_url = is_ecr_url
-
+types.ImageRepositoryType.transformer = types.ImageRepositoryType.Transformer(
+    converter=click.STRING,
+    transformation=is_ecr_url
+)
+
+def upload(self, image, resource_name):
+    try:
+        ecr_endpoint = get_registry_url(self)
+        if "amazonaws.com" in self.ecr_repo_multi.get(resource_name):
+            self.ecr_repo_multi[resource_name] = self.ecr_repo_multi[resource_name].replace("amazonaws.com", ecr_endpoint)
```

Comment:
small nit: could be slightly refactored, to fetch the entry from the dict only once, something like:
```suggestion
        ecr_repo = self.ecr_repo_multi.get(resource_name)
        if ecr_repo and "amazonaws.com" in ecr_repo:
            self.ecr_repo_multi[resource_name] = ecr_repo.replace("amazonaws.com", ecr_endpoint)
```

Maybe we could then also get rid of the `TypeError` in the `except` clause, or do you see any other situation where it could be raised?

Reply:
You are absolutely right.